src/secrets/components/SecretsTab.tsx

```diff
@@ -0,0 +1,178 @@
+}) => {
+  const [searchTerm, setSearchTerm] = useState<string>('')
+  const [sortDirection, setSortDirection] = useState<Sort>(Sort.Ascending)
+  const [sortKey, setSortKey] = useState<string>('id')
+  const [sortType, setSortType] = useState<SortTypes>(SortTypes.String)
+
+  const FilterSecrets = FilterList<Secret>()
+
+  const handleOpenCreateOverlay = () => {}
+
+  const createSecretButton = (
+    <Button
+      text="Create Secret"
+      color={ComponentColor.Primary}
+      icon={IconFont.Plus}
+      onClick={handleOpenCreateOverlay}
+      testID="button-create"
+    />
+  )
+
+  const SecretsEmptyState = (): JSX.Element => {
+    if (!searchTerm) {
+      return (
+        <>
+          <EmptyState size={ComponentSize.Medium}>
+            <EmptyState.Text>
+              Looks like there aren't any <b>Secrets</b>, why not create one?
+            </EmptyState.Text>
+            {createSecretButton}
+          </EmptyState>
+        </>
+      )
+    }
+
+    return (
+      <>
+        <EmptyState size={ComponentSize.Medium}>
+          <EmptyState.Text>No Secrets match your query</EmptyState.Text>
+        </EmptyState>
+      </>
+    )
+  }
+
+  const handleFilterChange = (searchTerm: string) => {
+    handleFilterUpdate(searchTerm)
+  }
+
+  const handleFilterUpdate = (searchTerm: string) => {
+    setSearchTerm(searchTerm)
+  }
+
+  const handleSort = (
+    sortKey: SecretSortKey,
+    sortDirection: Sort,
+    sortType: SortTypes
+  ): void => {
+    setSortDirection(sortDirection)
+    setSortKey(sortKey)
+    setSortType(sortType)
+  }
+
+  // const handleOpenImportOverlay = (): void => {
+  // const {history, match} = props
+  //
+  // history.push(`/orgs/${match.params.orgID}/settings/variables/import`)
+  // }
+
+  // const handleOpenCreateOverlay = (): void => {
+  // const {history, match} = props
+  //
+  // history.push(`/orgs/${match.params.orgID}/settings/variables/new`)
+  // }
+
+  // const handleDeleteSecret = (secret: Secret): void => {
+  // const {onDeleteSecret} = props
+  // onDeleteSecret(secret.id)
+  // }
+
+  const leftHeaderItems = (
+    <>
+      <SearchWidget
+        placeholderText="Filter secrets..."
+        searchTerm={searchTerm}
+        onSearch={handleFilterChange}
+      />
+      <ResourceSortDropdown
+        onSelect={handleSort}
+        resourceType={ResourceType.Secrets}
+        sortDirection={sortDirection}
+        sortKey={sortKey}
+        sortType={sortType}
+      />
+    </>
+  )
+
+  return (
+    <>
+      <TabbedPageHeader
+        childrenLeft={leftHeaderItems}
+        childrenRight={createSecretButton}
+      />
+      <GetResources resources={[ResourceType.Secrets]}>
+        <FilterSecrets
+          list={secrets}
+          searchTerm={searchTerm}
+          searchKeys={['key', 'secret[].key']}
+        >
+          {sc => (
+            <SecretsList
+              secrets={sc}
+              emptyState={SecretsEmptyState()}
+              onDeleteSecret={() => {}}
```

Comment:
the infrastructure seems to be in place for this, why isn't it being integrated?

Reply:
I started to and then realized as I was doing it that if I kept it up I'd end up doing the entire epic in one PR which is probably not the best of practices.  So I left things disconnected for now and I'll hook this up early next week.